java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -764,22 +785,27 @@ private void buildSwitchStatement(SwitchStatementTree switchStatementTree) {
     currentBlock = conditionBlock;
   }
 
-  private void buildSwitchExpression(SwitchExpressionTree tree) {
-    // FIXME When used as expression, switches are considered as a simple element
-    currentBlock.elements.add(tree);
+  /**
+   * A switch expression can use the traditional cases with 'column' (with fall-through) or,
+   * starting with java 12, the 'arrow' cases (without fall-through). Cases can not be mixed.
+   *
+   * @param switchExpressionTree the switch to evaluate
+   * @return true if the switch uses fall-through
+   */
+  private static boolean switchWithoutFallThrough(SwitchExpressionTree switchExpressionTree) {
+    return switchExpressionTree.cases().stream()
+      .map(CaseGroupTree::labels)
+      .flatMap(List::stream)
+      .noneMatch(CaseLabelTree::isFallThrough);
```

Comment:
I'm not sure to follow you. The thing is that `break` statements are going to be naturally connected, by construction, to adequate next block with the current implementation.

The new switch, however, is necessary without fall-through, while there is no `break`, and consequently we should force the connection to the end of the switch, and not the next case.

Reply:
problem is that `CaseLabelTree.isFallThrough` will return `true` for old style case with colon even if it is not falling through, because body contains `break`.  I think it would be better to use `isArrow` as a name for the function, as there is no ambiguity.